src/HoC.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import {ensure} from './util';
+import {update, unmount} from './Mixin';
```

Comment:
Semantically, I'd say that exporting these methods as part of the mixin default export is a bit misleading. These methods are also not tied directly to the mixin. The functions don't belong to the mixin any more than they belong to the HoC at this point, so It would probably make the most sense to put those methods in a different file entirely, like 'updateComponent.js' or something like that. Then both the HoC and mixin could pull from that file.